backend/Watchdog.Core/Watchdog.Core.BLL/Services/TeamService.cs

```diff
@@ -0,0 +1,126 @@
+using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Watchdog.Core.BLL.Services.Abstract;
+using Watchdog.Core.Common.DTO.Team;
+using Watchdog.Core.DAL.Context;
+using Watchdog.Core.DAL.Entities;
+
+namespace Watchdog.Core.BLL.Services
+{
+    public class TeamService : BaseService, ITeamService
+    {
+        public TeamService(WatchdogCoreContext context, IMapper mapper) : base(context, mapper) { }
+        
+        public async Task<ICollection<TeamDto>> GetAllTeamsAsync()
+        {
+            var teams = await _context.Teams
+                .Include(team => team.User)
+                .Include(team => team.Organization)
+                .Include(team => team.TeamMembers)
+                    .ThenInclude(teamMember => teamMember.Member)
+                    .ThenInclude(teamMember => teamMember.User)
+                .ToListAsync();
+            
+            return _mapper.Map<ICollection<TeamDto>>(teams);
+        }
+
+        public async Task<TeamDto> GetTeamAsync(int teamId)
+        {
+            var teamEntity = await _context.Teams
+                .Include(team => team.User)
+                .Include(team => team.Organization)
+                .Include(team => team.TeamMembers)
+                    .ThenInclude(teamMember => teamMember.Member)
+                    .ThenInclude(teamMember => teamMember.User)
+                .FirstOrDefaultAsync(team => team.Id == teamId);
+            
+            return _mapper.Map<TeamDto>(teamEntity);
+        }
+
+        public async Task<ICollection<TeamDto>> GetUserTeamsAsync(int userId)
+        {
+            var teams = await GetAllTeamsAsync();
+            var userTeams = teams
+                .Where(t => t.Members.Any(m => m.Id == userId))
+                .ToList();
+            
+            return userTeams;
+        }
+        
+        public async Task<ICollection<TeamDto>> GetNotUserTeamsAsync(int userId)
+        {
+            var teams = await GetAllTeamsAsync();
```

Comment:
I think you can combine these two methods

Reply:
Done.